cnf-certification-test/networking/netutil/netutil.go

```diff
@@ -77,3 +77,13 @@ func GetListeningPorts(cut *provider.Container) (map[PortInfo]bool, error) {
 
 	return parseListeningPorts(outStr)
 }
+
+func GetSSHDaemonPort(cut *provider.Container) (string, error) {
+	const findSSHDaemonPort = "grep -w 'Port' /etc/ssh/sshd_config"
```

Comment:
"[sshd](https://linux.die.net/man/8/sshd)(8) reads configuration data from /etc/ssh/sshd_config (**or the file specified with -f on the command line**)."
So, what if the sshd config was specified using the -f option?

Reply:
Could you show me a command with -f option?

Reply:
`sshd -f /home/user/.ssh/sshd_config` where `/home/user/.ssh/sshd_config` is a customized config file.

Reply:
Also the `-p port` option can be used to override the port specified in the `sshd_config` file.

Reply:
user specific sshd config can be used which can override global sshd config. But we cannot know which user is being used. Can we? We can just check if .ssh/sshd_config is present and get the Port from there.

Reply:
True, but could not we assume that the user is following best practices? Like keeping the sshd_config in its usual location, under .ssh folder in the home?

Reply:
I'm not sure that's a best practice. In any case I keep seeing weak points to this solution. Even if there's a `sshd_config` file in the user's home dir, you cannot be sure that the `-f` option was used with that file or that it was not overwritten with the `-p` option. In order to check this the command used to launch the ssh daemon could be analyzed. This would cover the use of a different port that 22 but not a different name for the daemon process.

Reply:
RH admins either set this config at global level or at the user's home dir .ssh folder, as per the common / best practices. If anywhere else is placed, there must be some reason, that I am not familiar with. But I agree, that case would not be covered.